src/odemis/driver/technolution.py

```diff
@@ -464,22 +757,28 @@ def __init__(self, detector, start_func, next_func, stop_func, get_func):
 
     def start_generate(self):
         """
-        Start the dataflow.
+        Start the dataflow using the provided function. The approriate settings are retrieved via the VA's of the
+        each component
         """
         self._start()
 
     def next(self, field_num):
+        """
+        Acquire the next field image using the provided function.
+        :param field_num (tuple): tuple with x,y coordinates in integers of the field 
+        :return: 
```

Comment:
remove?

Reply:
This method or all these methods by simply overwriting them directly?

Reply:
BTW, you could check right here that field_num looks fine. It should be at least a tuple with two ints >=0. (there are a few extra checks possible, such as it should be different from any previous value since starting the acquisition, and maybe even if it should be adjacent to a previous field, but that's up to you whether you want to write extra code to detect such error).

Reply:
This method does not return anything. So no return in the doc string.

@pieleric It would be very useful to have these checks. Or we could consider having those checks in the path planning, and here only that it needs to be two ints >= 0. That would give us some extra flexibility regarding the "reshoot" option in the path planner.

Reply:
Indeed, the checks are not going to be compatible with the reshoot functionality. Of course, you could also put this check in the path planning, but the idea was to check at the lower level. I'm sure the path planner will do the right thing.
An alternative would be to only check at the end of an acquisition: if the fields do no form a compact rectangle (ie, there are "holes"), then log a warning. 
Once we have reshoot functionality, I guess we hopefully are already convince the rest of the code works fine, so we can drop this check entirely ;-)